### PR TITLE
fix(model): convert bool mask_cache to float additive mask for softcapping

### LINKED ISSUE #1672: attention mask is incorrect when generate with softcapping
### Bug description

When using `litgpt generate` on models with softcapping, `build_mask_cache` creates mask as `torch.bool` 
https://github.com/Lightning-AI/litgpt/blob/ef9647cfa7cd73e03b0e29126bfe8b42cae509eb/litgpt/model.py#L465
and then it's added to scores.
https://github.com/Lightning-AI/litgpt/blob/ef9647cfa7cd73e03b0e29126bfe8b42cae509eb/litgpt/model.py#L309
Therefore, attention mask is not accounted for and it does +0 or +1 on scores.

### What operating system are you using?

Unknown

### LitGPT Version

``` 



```

### PR DESCRIPTION
**Pull request by nuthalapativarun**
## What does this PR do?

When the KV cache is active, `build_mask_cache()` returns a `torch.bool` tensor where `True` indicates a position that should be attended to (lower triangle). In `scaled_dot_product_attention`, for models that use `attention_logit_softcapping` (e.g. Gemma 2), this boolean mask was added directly to the softcapped scores:

```python
scores = scores + mask  # mask is torch.bool → adds 0 or 1, NOT 0 or -inf
```

This breaks causal masking: future positions received a score boost of `+1` instead of `-inf`, so softmax assigned non-zero attention weight to tokens that should be completely masked out.

### Fix

Add an `elif` branch that converts the incoming boolean mask to a float additive mask before the addition (`True → 0.0`, `False → -inf`). The same fix is applied to both `CausalSelfAttention` and `MultiheadLatentAttention`.

```python
elif mask.dtype == torch.bool:
    # build_mask_cache returns a boolean mask (True=keep); convert to additive float mask
    mask = torch.zeros_like(mask, dtype=q.dtype).masked_fill_(~mask, torch.finfo(q.dtype).min)
scores = scores + mask
```

### Testing

Added `test_attention_mask_bool_to_float_with_softcapping` which:
1. Verifies `mask_cache` is indeed `torch.bool` (pre-condition of the bug)
2. Runs a prefill forward pass with KV cache enabled
3. Runs the same forward pass without KV cache
4. Asserts the two outputs are numerically close — they diverge without the fix because the bool mask corrupts the attention distribution

Fixes #1672

### CODE REVIEW

**Comment by nuthalapativarun** [comment]
Hi! Just checking in — CI appears to be waiting on an authorized `/azp run` trigger. Happy to make any changes needed to move this forward. Thanks!
